Add auto-date histogram aggregations documentation

vagimeli opened the pull request:
### Description
Closes content gaps for the following bucket aggregations:  
Auto-interval date histogram
 Children
 Composite
 Parent
 Rare terms

### Issues Resolved
#4293 

### Version
Backport to latest version

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

vagimeli commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+
+To use the `auto_date_histogram` aggregation, you need to specify the field containing the date or timestamp values. Additionally, you can provide optional parameters to customize the behavior of the aggregation.
+
+For example, you can use the `auto_date_histogram` aggregation to :
```
```suggestion
For example, you can use the `auto_date_histogram` aggregation to create date-based buckets:
```

vagimeli commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,59 @@
+
+For example, you can use the `auto_date_histogram` aggregation to create date-based buckets:
+
+```json
````
Technical reviewer: Please confirm this example is relevant to an OpenSearch user. I tested the example using Dev Tools. If another example is more appropriate, please replace the draft example with your example. Thank you.

vagimeli commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+nav_order: 15
+---
+
+# Auto-date histogram
```
@msfroh @bowenlan-amzn @sandeshkr419 Would one of you be available to technical review this PR? The doc team is closing content gaps for aggregations and need a technical review before we can published the doc. Please note that I used internal tools to assist in code example generation and Dev Tools to test those examples. Thank you.

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+
+# Auto-date histogram
+
+The `auto_date_histogram` aggregation automatically creates date histogram buckets based on the time range of your data. This aggregation is particularly useful when you are working with time-series data and want to visualize or analyze data over different time intervals without having to specify the bucket size manually.
```
```suggestion
The `auto_date_histogram` aggregation automatically creates date histogram buckets based on the time range of your data. This aggregation is particularly useful when you are working with time-series data and want to visualize or analyze data over different time intervals without specifying the bucket size manually.
```

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+
+To use the `auto_date_histogram` aggregation, you need to specify the field containing the date or timestamp values. Additionally, you can provide optional parameters to customize the behavior of the aggregation.
+
+For example, you can use the `auto_date_histogram` aggregation to create date-based buckets:
```
```suggestion
For example, you can use the following `auto_date_histogram` aggregation to create date-based buckets:
```

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,59 @@
+```
+{% include copy-curl.html %}
+
+#### Example response
````
Instead of this heading, I would use a sentence describing what the response contains.

vagimeli replied:
@bowenlan-amzn Please help me address the information gap. Can you explain the response from a novice user point of view?

bowenlan-amzn replied:
If you can provide the data you have indexed, this can just be the response of the previous search request

vagimeli replied:
@bowenlan-amzn Please verify the revised text is technically accurate. Thank you.

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+      "auto_date_histogram": {
+        "field": "timestamp",
+        "buckets": 2,
+        "minimum_interval": "day",
```
What does this parameter specify?

vagimeli replied:
@bowenlan-amzn Please help me address the information gap. See comment above.

bowenlan-amzn replied:
The accepted units for minimum_interval are:

year
month
day
hour
minute
second

auto date histogram won't try the interval shorter than the one specified.

vagimeli replied:
@bowenlan-amzn Please confirm the revised text is technically accurate. Thank you.

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+        "field": "timestamp",
+        "buckets": 2,
+        "minimum_interval": "day",
+        "time_zone": "America/Los_Angeles",
```
What are the possible formats for `time_zone`?

vagimeli replied:
@bowenlan-amzn Please help me address the information gap. See comment above.

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,59 @@
+        "buckets": 2,
+        "minimum_interval": "day",
+        "time_zone": "America/Los_Angeles",
+        "format": "yyyy-MM-dd HH:mm:ss"
```
What are the possible formats for dates?

vagimeli replied:
@bowenlan-amzn Please help me address the information gap. See comment above.

bowenlan-amzn replied:
For any time data (date field), format and timezone are consistent. I would expect we have a different documentation page about these

kolchfa-aws reviewed (changes requested):
Thank you, @vagimeli! Can we add more info about these parameters? Is there granularity to the returned intervals? What are the formats for dates/times and time zones? Is there a default time zone?

@bowenlan-amzn Please provide technical input about the above questions. Where can I find these details?

vagimeli commented:
> Thank you, @vagimeli! Can we add more info about these parameters? Is there granularity to the returned intervals? What are the formats for dates/times and time zones? Is there a default time zone?
> 
> @bowenlan-amzn Please provide technical input about the above questions. Where can I find these details?

Hi @bowenlan-amzn @kolchfa-aws I've addressed feedback comments. Please review and approve at your availability. Thank you :)

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,377 @@
+
+## Returned buckets
+
+Each bucket of the aggregation contains the following information:
```
"bucket of the aggregation" => "aggregation bucket"?

kolchfa-aws replied:
I can remove "of the aggregation" altogether.

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+}
+```
+
+In OpenSearch, dates are internally stored as 64-bit integers representing timestamps in milliseconds since the epoch. In the aggregation response, each bucket `key` is returned as such a timestamp. The `key_as_string` value shows the same timestamp, but formatted as a date string based on the [`format`](#date-format) parameter. The `doc_count` field contains the number of documents in the bucket ...
````
```suggestion
In OpenSearch, dates are internally stored as 64-bit integers representing timestamps in milliseconds since the epoch. In the aggregation response, each bucket `key` is returned as such a timestamp. The `key_as_string` value shows the same timestamp but formatted as a date string based on the [`format`](#date-format) parameter. The `doc_count` field contains the number of documents in the bucket.
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,377 @@
+:--- | :--- | :--- 
+`field` | String | The field on which to aggregate. The field must contain the date or timestamp values. Either `field` or `script` is required.
+`buckets` | Integer | The desired number of buckets. The returned number of buckets is less than or equal to the desired number. Optional. Default is `10`.
+`minimum_interval` | String | The minimum interval to be used. Specifying a minimum interval can make the aggregation process more efficient. Valid values are `year`, `month`, `day`, `hour`, `minute`, `second`. Optional.
```
```suggestion
`minimum_interval` | String | The minimum interval to be used. Specifying a minimum interval can make the aggregation process more efficient. Valid values are `year`, `month`, `day`, `hour`, `minute`, and `second`. Optional.
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,377 @@
+`field` | String | The field on which to aggregate. The field must contain the date or timestamp values. Either `field` or `script` is required.
+`buckets` | Integer | The desired number of buckets. The returned number of buckets is less than or equal to the desired number. Optional. Default is `10`.
+`minimum_interval` | String | The minimum interval to be used. Specifying a minimum interval can make the aggregation process more efficient. Valid values are `year`, `month`, `day`, `hour`, `minute`, `second`. Optional.
+`time_zone` | String | Indicates to use a different time zone other than the default (UTC) for bucketing and rounding. You can specify the `time_zone` parameter as a [UTC offset](https://en.wikipedia.org/wiki/UTC_offset), such as `-04:00`, or an [IANA time zone ID](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones), such as `America/New_York`. Optional. Default is `UTC`. For more infor ...
```
```suggestion
`time_zone` | String | Specifies to use a time zone other than the default (UTC) for bucketing and rounding. You can specify the `time_zone` parameter as a [UTC offset](https://en.wikipedia.org/wiki/UTC_offset), such as `-04:00`, or an [IANA time zone ID](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones), such as `America/New_York`. Optional. Default is `UTC`. For more information, see [Time zone](#time-zone).
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+}
+```
+
+Now, specify a `time_zone` of `-02:00` 
````
```suggestion
Now, specify a `time_zone` of `-02:00`: 
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+}
+```
+
+The response contains two buckets in which the start time is shifted by two hours and starts with 23:00 on April 16, 2022:
````
```suggestion
The response contains two buckets in which the start time is shifted by 2 hours and starts at 23:00 on April 16, 2022:
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+```
+{% include copy-curl.html %}
+
+The response contains two 3-hour buckets, starting from midnight UTC on April 17, 2022:
````
```suggestion
The response contains two 3-hour buckets, starting at midnight UTC on April 17, 2022:
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+}
+```
+
+When using time zones with daylight saving time (DST) changes, the size of buckets near the transition may differ slightly from the size of neighboring buckets. 
````
```suggestion
When using time zones with daylight saving time (DST) changes, the size of a bucket that is near the transition may differ slightly from the size of a neighboring bucket. 
```

natebower commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+```
+
+When using time zones with daylight saving time (DST) changes, the size of buckets near the transition may differ slightly from the size of neighboring buckets. 
+{: .note}
````
Above: Does singular work here? Otherwise I think it needs to be "the sizes of buckets that are near the transition may differ slightly from the sizes of neighboring buckets."

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+```
+{% include copy-curl.html %}
+
+The response shows that the blog posts were aggregated into two buckets. The interval was automatically set to 1 year, with all three 2022 blog posts in one bucket and the 2023 blog post in another:
````
```suggestion
The response shows that the blog posts were aggregated into two buckets. The interval was automatically set to 1 year, with all three 2022 blog posts collected in one bucket and the 2023 blog post in another:
```

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
````diff
@@ -0,0 +1,377 @@
+```
+{% include copy-curl.html %}
+
+The response shows that the blog posts were aggregated into two buckets. The interval was automatically set to 1 year, with all three 2022 blog posts in one bucket and the 2023 blog post in another:
````
Added "collected"

kolchfa-aws commented on _aggregations/bucket/auto-interval-date-histogram.md:
```diff
@@ -0,0 +1,377 @@
+
+## Returned buckets
+
+Each bucket of the aggregation contains the following information:
```
```suggestion
Each bucket contains the following information:
```
